### PR TITLE
Don't use `exit` anymore for Html::redirect

### PR DESCRIPTION
**Pull request by Pierstoval**
Converts "exit" into an exception that will be converted by a proper redirection by the existing exception listener in the project

### CODE REVIEW

File: src/Glpi/Exception/Http/RedirectException.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+{
+    use HttpExceptionTrait;
+
+    public readonly string $url;
+
+    public function __construct(string $url, int $http_code = 302, string $message = '')
+    {
+        $this->url = $url;
+
+        // Will automatically trigger an exception if the HTTP code isn't a Redirect one.
```
Comment (cedric-anne) [inline src/Glpi/Exception/Http/RedirectException.php:53]:
```suggestion
        // Will automatically trigger an exception if the HTTP code isn't a Redirect one.
        // See `\Symfony\Component\HttpFoundation\RedirectResponse::isRedirect()`
```

File: src/Html.php [hunk 2]
```diff
@@ -430,32 +431,12 @@ public static function back(): never
      * @param string $http_response_code Forces the HTTP response code to the specified value
      *
      * @return never
+     *
+     * @TODO: check if $http_response_code is still used.
```
Comment (cedric-anne) [inline src/Html.php:435]:
```suggestion
```

File: src/Glpi/Exception/Http/RedirectException.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * @internal Not to be used unless you absolutely know what you are doing. This will be removed in the future when all the code is under Dependency Injection and proper HTTP routing.
+ */
+class RedirectException extends BaseException implements HttpExceptionInterface
+{
+    use HttpExceptionTrait;
+
+    public readonly string $url;
+
+    public function __construct(string $url, int $http_code = 302, string $message = '')
```
Comment (cedric-anne) [inline src/Glpi/Exception/Http/RedirectException.php:49]: There is no reason to have a custom message. Should we keep this parameter ?

```suggestion
    public function __construct(string $url, int $http_code = 302)
```

**Review by cedric-anne — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
Seems like the original code was able to handle some specific case where a redirect response was not enough (header sent and ajax calls).

Not sure how this is used and removing it will probably break things :/

**Review by trasher — APPROVED** [review]

**Comment by Pierstoval** [comment]
> header sent and ajax calls

If headers are already sent and you want to redirect, this means that the process is already buggy. Cannot redirect when headers already sent anyway.
For AJAX calls,

**Pull request merged**